src/Quartz.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -31,91 +42,75 @@ public static class ServiceCollectionExtensions
 
                 return LogProvider.CurrentLogProvider as MicrosoftLoggingProvider;
             });
-
-            var builder = new ServiceCollectionQuartzConfigurator(services, SchedulerBuilder.Create());
-            configure?.Invoke(builder);
-
+            
             // Note that we can't call UseSimpleTypeLoader(), as that would overwrite any other configured type loaders
             services.TryAddSingleton(typeof(ITypeLoadHelper), typeof(SimpleTypeLoadHelper));
 
-            services.AddSingleton<ISchedulerFactory>(serviceProvider =>
+            var schedulerBuilder = SchedulerBuilder.Create(properties);
+            if (configure != null)
             {
-                // try standard appsettings.json
-                var config = serviceProvider.GetService<IConfiguration>();
-                var section = config.GetSection("Quartz");
-                var options = new NameValueCollection();
-
-                foreach (var kvp in section.GetChildren())
-                {
-                    options.Set(kvp.Key, kvp.Value);
-                }
-
-                // now override with programmatic configuration
-                foreach (string? key in builder.Properties.Keys)
+                var target = new ServiceCollectionQuartzConfigurator(services, schedulerBuilder);
+                configure(target);
+            }
+            
+            services.Configure<QuartzOptions>(options =>
+            {
+                foreach (var key in schedulerBuilder.Properties.AllKeys)
                 {
-                    options.Set(key, builder.Properties[key]);
+                    options[key] = schedulerBuilder.Properties[key];
                 }
-
-                return new ServiceCollectionSchedulerFactory(serviceProvider, options);
             });
-            return services;
-        }
+            
+            services.AddSingleton<ContainerConfigurationProcessor>();
```

Comment:
Should you use `TryAdd` here to avoid re-registering these days if `AddQuartz()` is called multiple times? It would be great if `AddQuartz()` was fully idempotent.

Reply:
Changed to use TryAdd